### DISCUSSION TITLE
Using custom slot in query template

### DISCUSSION THREAD

**Discussion by DonKirkham** [microsoft-search/pnp-modern-search#4174] [Q&A] [answered]
I apologize if this has been addressed, but I didn't find it in search.

In the Search Results webpart, can I use a custom slot in the query template? ChatGPT say yes, with the syntax {?customSlotName} and Copilot says yes with a syntax of {customSlotName}, but neither seem to work.

Scenario: I have a customSlot named MyCategory. Its value is a string, e.g., "Technology." I use this custom slot in my display template in several places. I also have the following in my query template:  Category:"Technology"
I would like to use the custom slot in the query template, e.g., Category:{MyCategoy} but that is not working with either syntax listed above.  
While updating the webpart in 2 places is not unreasonable, my goal is to reuse this webpart in many places in multiple sites so I want to make it as easy as possible for users to change 1 or 2 parameters to get the desired result.
I tried just pulling the value directly from the query template to display, but the query template has several elements and using handlebars to "build" a simple string from the query template is very inefficient. The full query template is:
-Title:"Other"
NavigatorOWSBOOL: 1
ArchiveOWSBOOL: 0
path:{Site.URL}/Lists/NavigatorChoices
Category:{MyCategory} 

In this example, {Site.URL} work perfectly. I would like to do the same with {MyCategory}

Thanks! -Don

**Comment by kasperbolarsen** [comment, marked as answer]
Hi Don,
It sounds like you are looking for something like this:
 <span class="pnp-QuickLinkBox-text">{{slot item @root.slots.QuickLinkBoxText}}</span>
where QuickLinkBoxText is one of my custom Slots.

You can find additional Layout samples as https://github.com/microsoft-search/pnp-modern-search-layouts/tree/main/Results/Handlebars/ for inspiration :-)

**Comment by DonKirkham** [comment]
Thanks Kasper, that worked perfectly.  I love community!

________________________________
